src/js/setup-text-tracks.js

```diff
@@ -0,0 +1,136 @@
+import dashjs from 'dashjs';
+
+/*
+ * Attach text tracks from dash.js to videojs
+ *
+ * @param {videojs} player the videojs player instance
+ * @param {array} tracks the tracks loaded by dash.js to attach to videojs
+ *
+ * @private
+ */
+function attachDashTextTracksToVideojs(player, tracks) {
+  const mediaPlayer = player.dash.mediaPlayer;
+
+  const trackDictionaryKeyedByTextTrack = new WeakMap();
+
+  // Add remote tracks
+  tracks
+    // Map input data to match HTMLTrackElement spec
+    // https://developer.mozilla.org/en-US/docs/Web/API/HTMLTrackElement
+    .map((track) => Object.assign(
+      {},
+      track,
+      {
+        default: track.defaultTrack,
+        kind: track.kind,
+        label: track.lang,
+        srclang: track.lang,
+      }
+    ))
+
+    // Add track to videojs track list
+    .map((track) => {
+      const remoteTextTrack = player.addRemoteTextTrack(track, true);
+      trackDictionaryKeyedByTextTrack.set(remoteTextTrack.track, track);
+
+      // Don't add the cues becuase we're going to let dash handle it
+      // natively. This will ensure that dash will stream new text tracks if
+      // necessary because video.js will only render tracks it has stored
+      // locally. Dash is capable of loading separate streams for subtitles
+      //
+      // Example:
+      // https://storage.googleapis.com/shaka-demo-assets/sintel-mp4-wvtt/dash.mpd
+
+      // Return `track` so we can continue chaning.
+      return track;
+    })
+  ;
+
+  /*
+   * Scan `videojs.textTracks()` to find one that is showing. Set the dash
+   * text track.
+   */
+  function updateDashTextTrack() {
+    const dashMediaPlayer = this.dash.mediaPlayer;
+    const textTracks = this.textTracks();
+    let activeTextTrackIndex = -1;
+
+    function findMatchingTrack(dashTrack) {
```

Comment:
That was my original plan, but jshint didn't like me creating functions within a loop.